services/config/config.go

```diff
@@ -0,0 +1,109 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package config provides configuration facility.
+package config
+
+import (
+	_ "embed"
+	"io/ioutil"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+
+	"github.com/percona/pmm-managed/services/platform"
+	"github.com/percona/pmm-managed/services/telemetry"
+)
+
+const (
+	envConfigPath     = "PERCONA_PMM_CONFIG_PATH"
+	defaultConfigPath = "/etc/percona/pmm/pmm-managed.yml"
```

Comment:
who and how creates this conf file ? are there changes to pmm-update or pmm-server?

Reply:
At the moment it is an extension point that we can use in the future. At the moment it can be used while developing (e.g. in devcontainer) via docker volume mount.

The idea is to have options to override config on different levels:
1. env variable pointing to a config file
2. default location of a config file
3. embedded config

1 - highest priority